src/bundle/Security/Http/EventListener/CheckTwoFactorCodeReuseListener.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Scheb\TwoFactorBundle\Security\Http\EventListener;
+
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Psr\Log\LoggerInterface;
+use Scheb\TwoFactorBundle\Security\TwoFactor\Event\TwoFactorCodeCheckEvent;
+use Scheb\TwoFactorBundle\Security\TwoFactor\Event\TwoFactorCodeReusedEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use function sha1;
+
+/**
+ * @final
+ */
+class CheckTwoFactorCodeReuseListener implements EventSubscriberInterface
+{
+    public const LISTENER_PRIORITY = 0;
+
+    public function __construct(
+        private readonly EventDispatcherInterface $eventDispatcher,
+        private readonly mixed $cache,
+        private readonly int $cacheDuration,
+        private readonly LoggerInterface|null $logger,
+    ) {
+    }
+
+    public function checkForCodeReuse(TwoFactorCodeCheckEvent $event): void
+    {
+        if (null === $this->cache) {
+            return;
+        }
+
+        $cacheItem = null;
+        $cacheKey = 'scheb_two_factor_code_reuse.'
+            .sha1($event->getUser()->getUserIdentifier().'.'.$event->getCode());
+
+        if ($this->cache instanceof CacheItemPoolInterface) {
+            $cacheItem = $this->cache->getItem($cacheKey);
+            $cacheItem->expiresAfter($this->cacheDuration);
+            $cacheItem->set(true);
+            $this->cache->save($cacheItem);
+        }
+
+        if (!$cacheItem instanceof CacheItemInterface) {
```

Comment:
I believe this should be the `else` branch, executed whenever `$this->cache` is not a supported interface.

```suggestion
        } else {
```

Reply:
That is a leftover from when I checked for 2 different interfaces and will be changed!